### ISSUE TITLE
Cannot overload show on Julia v0.5

### ISSUE THREAD

**Issue by dpsanders** [SimonDanisch/FixedSizeArrays.jl#95]
The following came up in `ValidatedNumerics.jl`:

```
julia> using FixedSizeArrays

julia> immutable Hello{N, T} <: FixedVector{N, T}
           a :: NTuple{N, T}
       end

julia> Base.show(io::IO, x::Hello) = println(io, "HELLO")

julia> x = Hello(1,2)
Hello(1,2)
```

i.e. it seems to be using the generic `show` from `FixedSizeArrays`, not the newly overloaded one for my new type.

This _only_ happens on Julia v0.5, so I am guessing it's actually a Julia problem?

**Comment by dpsanders** [comment]
This seems to be due to the fact that `FixedSizeArrays` overloads the `showcompact` function. Is there any problem with changing this to just `show` instead? If not, I will submit a (trivial) PR.

**Comment by dpsanders** [comment]
This (only overloading `show` in `FixedSizeArrays`) gives the following output:

```
julia> using FixedSizeArrays

julia> immutable Hello{N, T} <: FixedVector{N, T}
                  a :: NTuple{N, T}
              end

julia> v = [Hello(i, i+1) for i in 1:100]
100-element Array{Hello{2,Int64},1}:
 Hello(1,2)
 Hello(2,3)
 Hello(3,4)
 Hello(4,5)
 Hello(5,6)
 Hello(6,7)
 Hello(7,8)
 Hello(8,9)
 Hello(9,10)
 Hello(10,11)
 ⋮
 Hello(91,92)
 Hello(92,93)
 Hello(93,94)
 Hello(94,95)
 Hello(95,96)
 Hello(96,97)
 Hello(97,98)
 Hello(98,99)
 Hello(99,100)
 Hello(100,101)
```

**Comment by SimonDanisch** [comment]
that looks good :)

**Comment by dpsanders** [comment]
Fixed in #96.
